packages/babel-parser/src/plugins/flow.js

```diff
@@ -1188,6 +1216,19 @@ export default (superClass: Class<Parser>): Class<Parser> =>
       const start = property.start;
       const length =
         property.value.params.length + (property.value.rest ? 1 : 0);
+
+      if (property.value.this && property.kind === "get") {
```

Comment:
nit: I feel like it is more intuitive if we wrap these errors into a single branch:
```js
if (property.value.this) {
  this.raise(
    property.value.this.start,
    property.kind === "get" ? FlowErrors.GetterMayNotHaveThisParam : FlowErrors.SetterMayNotHaveThisParam
  )
}
```